### ISSUE TITLE
Introduce source code vulnerability scanning 

### ISSUE THREAD

**Issue by levivannoort** [amplication/amplication#4539]
To ensure compliance with SOC 2, one of the concepts that needs implementation is the scanning of source code for vulnerabilities. 

> SOC.77 'Source code vulnerability scans are performed'
> "Automated source code vulnerability scans are performed for each merge to default branches in an attempt to identify and fix security-related weaknesses (flaws) in the code." 

Proof to deliver are 'Source code vulnerability scan reports presenting found issues' and 'Source code vulnerability scan configuration'.

**Comment by levivannoort** [comment]
After some internal discussion and a proof of concept we'll start by setting up source code vulnerability scanning via Trivy (https://github.com/aquasecurity/trivy). Which uses the package-lock.json when executing file system scanning, where based on specific versions of dependencies vulnerabilities are detected. It also allows to push the findings to github security which creates an alert per finding with information regarding the vulnerability and a possible solution.

e.g. vulnerability finding during proof of concept:
![Screenshot 2022-12-05 at 16 45 30](https://user-images.githubusercontent.com/73097785/205680026-6cd770e8-0c71-4bea-904b-1e4964952841.png)

**Comment by levivannoort** [comment]
Things to investigate:
- Running the vulnerability scan on push/pull request and have it need to pass as a check - and if it fails have a table output of the alert(s)

**Comment by levivannoort** [comment]
@overbit Let's further discus what we want to deliver functionality wise before starting the soc2 audit. I think the current implementation (see development branch: https://github.com/amplication/amplication/tree/4539-introduce-source-code-vulnerability-scanning) would suffice - i.e. adding additional vulnerability scanning and misconfiguration (e.g. dockerfiles) to the repository.

As we're currently in the process of refactoring the pipelines anyway it wouldn't make sense to sink in the time and also we probably need to weed out the false positives/other errors before presenting it as a check to our users.

**Comment by levivannoort** [comment]
Pull request was openend and reviewed. For now the trivy scan co-exist with the dependebot workflow, they shouldn't interfere, we could maybe look at disabling the dependabot of the trivy scan gives the exact same feedback, otherwise we might have to supplment it with additional type of scanning/config.

**Comment by levivannoort** [comment]
@overbit as the SOC2 control specifies "Automated source code vulnerability scans are performed for each merge to default branches in an attempt to identify and fix security-related weaknesses (flaws) in the code.", I think our current workflows suffice. They are ran everytime a pull-request is opened, et cetera and scan based on used packages whether a vulnerability is present. Do you still want to add the CodeQL for SAST?

**Comment by overbit** [comment]
I'd suggest completing this PR as it is and raising a new issue for the SAST implementation.

**Comment by levivannoort** [comment]
https://github.com/amplication/amplication/security/code-scanning?query=is%3Aopen+branch%3A4539-introduce-source-code-vulnerability-scanning++
